forum/qa-plugin/report-reason/q2apro-flag-reasons-page.php

```diff
@@ -0,0 +1,209 @@
+        $reasonId   = (int) $flagData['reasonId'];
+
+        $notice = empty($flagData['notice']) ? null : trim($flagData['notice']);
+        $userId = qa_get_logged_in_userid();
+
+        require_once QA_INCLUDE_DIR . 'app/votes.php';
+        require_once QA_INCLUDE_DIR . 'app/posts.php';
+        require_once QA_INCLUDE_DIR . 'pages/question-view.php';
+
+        $processingFlagReasonError = $this->processFlag($postType, $userId, $postId, $questionId, $reasonId, $notice);
+
+        $reply = $processingFlagReasonError ?
+            $this->wrapFlagReasonError($processingFlagReasonError) :
+            ['newFlags' => $this->wrapOutput(q2apro_count_postflags_output($postId), $postType, $postId)];
+
+        echo json_encode($reply);
+    }
+
+    private function processFlag($postType, ...$flagParams) {
+        $processingFlagReasonError = '';
+
+        switch ($postType) {
+            case 'q': {
+                $processingFlagReasonError = call_user_func_array([$this, 'processFlagToQuestion'], $flagParams);
+                break;
+            }
+            case 'a': {
+                $processingFlagReasonError = call_user_func_array([$this, 'processFlagToAnswer'], $flagParams);
+                break;
+            }
+            case 'c': {
+                $processingFlagReasonError = call_user_func_array([$this, 'processFlagToComment'], $flagParams);
+                break;
+            }
+            default: {
+                $processingFlagReasonError = 'UNRECOGNIZED_POST_TYPE:' . $postType;
+            }
+        }
+
+        return $processingFlagReasonError;
+    }
+
+    private function processFlagToQuestion($userId, $postId, $questionId, $reasonId, $notice) {
+        list($question, $childPosts, $aChildPosts, $closePost, $duplicatePosts) = qa_db_select_with_pending(
+            qa_db_full_post_selectspec($userId, $questionId),
+            qa_db_full_child_posts_selectspec($userId, $questionId),
+            qa_db_full_a_child_posts_selectspec($userId, $questionId),
+            qa_db_post_parent_q_selectspec($questionId),
+            false
+        );
+
+        $questionFlagError = qa_flag_error_html($question, $userId, $questionId);
+        if ($questionFlagError) {
+            return $questionFlagError;
+        }
+
+        $answers = qa_page_q_load_as($question, $childPosts);
+        $commentsFollows = qa_page_q_load_c_follows($question, $childPosts, $aChildPosts, $duplicatePosts);
+
+        if (qa_flag_set_tohide($question, $userId, qa_userid_to_handle($userId), qa_cookie_get(), $question)) {
+            qa_question_set_status(
+                $question, QA_POST_STATUS_HIDDEN, null, null, null, $answers, $commentsFollows, $closePost
+            ); // hiding not really by this user so pass nulls
+        }
+
+//        var_dump('<br>$reasonId <= $this->CUSTOM_REPORT_REASON_ID: ', $reasonId <= $this->CUSTOM_REPORT_REASON_ID, ' /$this->CUSTOM_REPORT_REASON_ID: ', $this->CUSTOM_REPORT_REASON_ID);
```

Comment:
to chyba jednak do usunięcia

Reply:
Docelowo tak - zostawiłem niektóre `var_dump` i `echo`, bo spodziewałem się, że będę do tych miejsc jeszcze wracał w celu np. poprawy walidacji.